src/wp-includes/nav-menu.php

```diff
@@ -780,6 +759,35 @@ function wp_get_nav_menu_items( $menu, $args = array() ) {
 	return apply_filters( 'wp_get_nav_menu_items', $items, $menu, $args );
 }
 
+/**
+ * @param $items
+ *
+ */
+function _prime_menu_item_objects( $items ){
```

Comment:
Naming things:

I suggest `menu_items` and `menu_item` in the `foreach` loop. 

This will make it clearer to future developers (possibly you or me) that the `get_post_meta` calls are against the already cached menu items rather that the object they refer to. A comment to the affect would be handy too.